### PR TITLE
Boolean bug fix

### LINKED ISSUE #3318: [BUG] _Replace With Suitable Title_
**Summary**

Undefined behavior when Boolean expression in `if` command is single non-numeric text string, no spaces, no quotes
 
**LAMMPS Version and Platform**

Mac OS

**Expected Behavior**

Exit with error message `ERROR: Invalid Boolean syntax in if command` or `ERROR: Invalid syntax in variable formula`

**Actual Behavior**

Evaluates as either TRUE or FALSE, I have seen both on consecutive tries

**Steps to Reproduce**

./lmp
if (False) then 'print "True"'

**Further Information, Files, and Links**

### PR DESCRIPTION
**Pull request by sjplimp**
**Summary**

Fix corner case with if-then-else Boolean logic where there is just a single string argument.  Also remove doc
page references to comparing strings with Boolean operaters like "<" or ">=".  This is not supported in the code.
The "==" and "!=" operators are supported for strings or numbers.

**Related Issue(s)**

Closes #3318

**Author(s)**

Steve

**Licensing**

By submitting this pull request, I agree, that my contribution will be included in LAMMPS and redistributed under either the GNU General Public License version 2 (GPL v2) or the GNU Lesser General Public License version 2.1 (LGPL v2.1).

**Backward Compatibility**

**Implementation Notes**

Code will now throw some errors which it didn't before.  See summary.

**Post Submission Checklist**

- [x] The feature or features in this pull request is complete
- [x] Licensing information is complete
- [x] Corresponding author information is complete
- [x] The source code follows the LAMMPS formatting guidelines
- [x] Suitable new documentation files and/or updates to the existing docs are included
- [x] The added/updated documentation is integrated and tested with the documentation build system
- [x] The feature has been verified to work with the conventional build system
- [x] The feature has been verified to work with the CMake based build system
- [x] Suitable tests have been added to the unittest tree.
- [ ] A package specific README file has been included or updated
- [ ] One or more example input decks are included

**Further Information, Files, and Links**

### CODE REVIEW

**Review by athomps — COMMENTED** [review]
Please change result for single string to ERROR, not FALSE

**Comment by sjplimp** [comment]
@athomps made the change

**Review by athomps — APPROVED** [review]
I approve.

**Comment by akohlmey** [comment]
@sjplimp the changes needed to be reflected in the corresponding unit tests. I have done that and also added some more tests to catch the new errors and additional special cases discussed.

File: src/variable.cpp [hunk 1]
```diff
@@ -4797,32 +4799,40 @@ double Variable::evaluate_boolean(char *str)
             delete[] str1;
             delete[] str2;
           }
+
         } else if (opprevious == LT) {
-          if (flag2) error->all(FLERR,"Invalid Boolean syntax in if command");
-          if (value1 < value2) argstack[nargstack].value = 1.0;
```
Comment (akohlmey) [inline src/variable.cpp:4802]: @sjplimp this change introduces a bug (`$a < $b` is always false) that was flagged in extended regression testing. Will submit a PR with the fix (and a few others).

Comment (sjplimp) [reply src/variable.cpp:4802]: oops - must have whacked that line for LT when refactoring - good catch

**Pull request merged**